pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -339,6 +351,21 @@ public String getTaskType() {
           .info("Finished generating task configs for table: {} for task: {}, numTasks: {}", offlineTableName, taskType,
               pinotTaskConfigsForTable.size());
     }
+
+    // Reset watermarks for invalid tables. This covers the metrics clean up when the table is removed or the merge
+    // config is added and then removed.
+    LeadControllerManager leadControllerManager = _clusterInfoAccessor.getLeaderControllerManager();
```

Comment:
L385-L268 can be changed as
```
for (String tableNameWithType : _mergeRollupWatermarks.keySet()) {
    if (!leadControllerManager.isLeaderForTable(tableNameWithType)) {
        resetWatermarkMs(tableNameWithType);
    }
}
```
Since _mergeRollupWatermarks is a HashMap, so looping over _mergeRollupWatermarks.keySet() will not iterating duplicate keys, so the `!candidateMergeTables.contains(tableNameWithType)` check is not needed.

Reply:
But, we still need to call `resetWatermark()` for tables that shows up in the `_mergeRollupWatermarks` but no longer be part of `candidateMergeTables (merge enabled tables)`.

Reply:
@mqliang I changed the logic and put `cleanUpMetrics()` function. Can you take a look again?